lib/logproto/tests/test-text-server.c

```diff
@@ -589,3 +591,33 @@ Test(log_proto, test_log_proto_text_server_io_eagain)
 
   log_proto_server_free(proto);
 }
+
+Test(log_proto, buffer_split_with_encoding_and_position_tracking)
+{
+  GString *data = g_string_new("Lorem ipsum\xe2\x98\x83lor sit amet, consectetur adipiscing elit\n");
+  GString *data_smaller = g_string_new("muspi merol\n");
+  gsize bytes_should_not_fit = 5;
+
+  proto_server_options.max_msg_size = data->len + data_smaller->len - bytes_should_not_fit;
+  proto_server_options.max_buffer_size = proto_server_options.max_buffer_size;
```

Comment:
I think this is a typo in the initialization: `max_buffer_size = max_buffer_size` is a self-assignment.

Reply:
Nice catch, thanks. I'll fix it.